unleash-yggdrasil/src/strategy_parsing.rs

```diff
@@ -327,6 +330,30 @@ fn rollout_constraint(mut node: Pairs<Rule>) -> RuleFragment {
     })
 }
 
+fn get_hostname() -> Result<String, SdkError> {
+    //This is primarily for testing purposes
+    if let Ok(hostname_env) = env::var("HOSTNAME") {
```

Comment:
Hard to do an automated test in any other way since the hostname crate does an OS call rather than an env var read. Validated locally against my own machine though